core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -620,6 +623,8 @@ class DatabricksAwsPlatform(gpuDevice: Option[GpuDevice],
     Some(NodeInstanceMapKey(instanceType = "g5.8xlarge"))
   }
 
+  override def maxGpusSupported: Int = 4
```

Comment:
why 4? It is  node-dependent, right? should it come from the NodeInstance definitions?
at least lets have a comment to reference where is this 4 is coming from.

Reply:
Updated the logic to dynamically use the max value from the instance type map for the given platform else default to 1.